spec/lib/wakes/redirect_mapper_spec.rb

```diff
@@ -174,8 +174,61 @@
     it 'creates a new resource' do
       described_class.new '/source', '/target', 'New Label'
 
-      expect(Wakes::Location.find_by(:path => '/target').resource)
+      expect(Wakes::Location.find_by(:host => nil, :path => '/target').resource)
         .to have_wakes_graph(:canonical_location => '/target', :legacy_locations => ['/source'])
     end
   end
+
+  describe 'handles non-default hosts' do
+    context 'source is default, target is default' do
+      it 'points the source location to the target resource' do
+        described_class.new '/source', '/target', 'New Label'
+
+        expect(Wakes::Location.find_by(:host => nil, :path => '/target').resource)
+          .to have_wakes_graph(:canonical_location => '/target', :legacy_locations => ['/source'])
+      end
+    end
+
+    context 'source is default, target is non-default' do
+      it 'points the source location to the target resource' do
+        described_class.new '/source', '/target', 'New Label', nil, 'some.host'
```

Comment:
following the other comment about API, this would be a lot nicer if it was `describe_class.new '/source', 'some.host/target', :label => 'New Label'`